File: 2839/2839.cpp
/*
Title   : BOJ_2839 [설탕 배달]
Author  : Hoseok Lee
Date    : 2021/03/09

https://www.acmicpc.net/problem/2839
https://github.com/hoshogi
*/

#include <iostream>
using namespace std;

int main(void)
{
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int n, count = 0;
    
    cin >> n;
    
    /*
    먼저 5로 나누어 지는지 확인한다.
    5로 나누어 지지않으면 3씩 빼고 count에 1을 추가하면서 확인한다.
    */
    
    while (true)
    {
        if (n % 5 == 0)
        {
            count += n / 5;
            n = 0;
        }
        else
        {
            n -= 3;
            count++;
        }
        
        if (n == 0)
        {
            cout << count << '\n';
            break;
        }
        else if (n < 0)
        {
            cout << -1 << '\n';
            break;
        }
    }
    
    return 0;
}

File: 2839/2839.java
/*
Title   : BOJ_2839 [설탕배달]
Author  : Hoseok Lee
Date    : 2024/02/13

https://www.acmicpc.net/problem/2839
https://github.com/hoshogi
*/

import java.io.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int sugarWeight = Integer.parseInt(br.readLine());
        int sugarBag = sugarWeight / 5;
        int leftSugar = sugarWeight % 5;

        while (leftSugar != 0) {
            if (leftSugar % 3 == 0) {
                sugarBag += leftSugar / 3;
                break;
            }

            sugarBag--;
            leftSugar += 5;

            if (leftSugar > sugarWeight) {
                sugarBag = -1;
                break;
            }
        }

        bw.write(Integer.toString(sugarBag));
        bw.flush();
        bw.close();
    }
}